packages/ci/src/lib/issues.unit.test.ts

```diff
@@ -271,4 +275,79 @@ describe('issues sorting', () => {
       ),
     ).toBe(0.09); // 1% + 8% = 9%
   });
+
+  it('should return 0 when there are no categories', () => {
+    expect(
+      getAuditImpactValue(
+        {
+          audit: {
+            slug: 'react-jsx-key',
+            title: 'Disallow missing `key` props in iterators',
+          },
+          plugin: { slug: 'eslint', title: 'ESLint' },
+        },
+        {
+          plugins: [
+            {
+              slug: 'eslint',
+              groups: [
+                {
+                  slug: 'suggestions',
+                  refs: [{ slug: 'mock-rule', weight: 1 }],
+                },
+              ],
+            },
+          ],
+        } as Report,
+      ),
+    ).toBe(0);
+  });
+});
+
+describe('calculateGroupImpact', () => {
+  const mockAudit = {
+    slug: 'react-jsx-key',
+    title: 'Disallow missing `key` props in iterators',
+  };
+  const mockCategoryRef = {
+    type: 'group',
+    plugin: 'eslint',
+    slug: 'suggestions',
+    weight: 1,
+  } as CategoryRef;
+
+  const mockReport = {
+    plugins: [
+      {
+        slug: 'eslint',
+        groups: [
+          {
+            slug: 'suggestions',
+            refs: [
+              ...Array.from({ length: 9 }).map((_, i) => ({
+                slug: `mock-rule-${i}`,
+                weight: 1,
+              })),
+              { slug: 'react-jsx-key', weight: 1 },
+            ],
+          },
+        ],
+      },
+    ],
+  } as Report;
+
+  it('should calculate correct impact for audit in group', () => {
+    expect(calculateGroupImpact(mockCategoryRef, mockAudit, mockReport)).toBe(
+      0.1,
+    );
+  });
+
+  it('should return 0 if the group has no refs', () => {
+    const emptyGroupRef = {
+      ...mockCategoryRef,
+      plugin: 'eslint',
+      slug: 'nonexistent-group',
+    };
+    expect(calculateGroupImpact(emptyGroupRef, mockAudit, mockReport)).toBe(0);
+  });
```

Comment:
Empty refs isn't really a realistic scenario, our schema validations demand a non-empty array. The `!group?.refs.length` condition is more concerned with `group` being `undefined`, i.e. when no matching plugin and group could be found.

Reply:
Interesting 🤔 `!group?.refs.length` check was a bit misleading, but now I get it